content/blog/devtools_ui.md

```diff
@@ -0,0 +1,108 @@
+
+After some useful discussions within our teams and even talking to the Chrome DevTools Team we reached the conclusion that we needed to build a completely independent webapp for DevTools UI, which can be later integrated in our projects and OSS can leverage it somehow.
+
+The final solution was crafted with the help of a bunch of tools we use at Sauce Labs:
+A Gitlab project consisting of several build scripts like fetching required dependencies to compile DevTool UI and keeping the built artifacts in gitlab to be used in our CI/CD pipelines later.
+
+Here is an excerpt of one of our recipes to compile artifacts. In the first part, this script fetches depot tools and then clones the DevTools frontend repository. In the second part, it generated all the required artifacts that are needed to serve Chrome DevTools UI:
+
+```bash
+#!/bin/bash
+
+## cloning scripts by Chromium to build chrome devtools ui
+git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
+export PATH="$PATH:$(pwd)/depot_tools"
+mkdir devtools && cd devtools
+fetch devtools-frontend
+cd devtools-frontend
+
+## building artifacts
+gn gen out/Default
+autoninja -C out/Default
+```
+
+All artifacts are stored in JFROG Artifactory with proper versioning as the Chromium team doesn't have any versioning scheme for it. This is quite helpful in order to roll back to a specific version. The final step is to push stored artifacts in dedicated AWS S3 buckets for DevTools UI artifacts in different data centres to alleviate latency issues to our customers over dedicated Cloudfront Distributions. A checksum technique is applied in order to speed up our CI/CD to use artifacts within Gitlab (stored in the first stage) instead of fetching from our artifactory repository and push to S3 bucket.
+
+The best part of this solution is to ship changes to customers within a few clicks in the CI/CD.
+
+{{< figure src="/images/blog/devtools-ui/pipeline.gif" class="img-responsive text-center" >}}
+
+Here we have an overview of the architecture:
+
+{{< figure src="/images/blog/devtools-ui/architecture.gif" class="img-responsive text-center" >}}
+
+__Final Step:__ To integrate this UI into any webapp, all  you need is pass your WebSocket debug URL of a webpage as a query param to establish connection between DevTools and your application.
+
+> [https://d2rso8miivgcsi.cloudfront.net/inspector.html/devtools-ui/inspector.html?wss=${webSocketUrl}](https://d2rso8miivgcsi.cloudfront.net/inspector.html/devtools-ui/inspector.html?wss=${webSocketUrl})
```

Comment:
This is staging CDN which has some AWS permission issues. 
Can we use here Production CDN which will be exposed to the users later.
`https://da38tdtjguzrq.cloudfront.net/inspector.html?wss=${webSocketUrl}`